src/test/java/com/neo4j/docker/neo4jserver/TestHealthcheck.java

```diff
@@ -0,0 +1,112 @@
+                 .withEnv( confNames.get( Setting.HTTP_LISTEN_ADDRESS ).envName, listenAddress )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+        return container;
+    }
+
+    private GenericContainer createContainerWithConfigListenAddress( String listenAddress ) throws IOException
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+
+        var tempConfigDir = temporaryFolderManager.createTempFolder( "temp_neo4j_config" );
+
+        var neo4jConfig = new File( tempConfigDir.toAbsolutePath() + File.separator + "neo4j.conf" );
+        var writer = new BufferedWriter( new FileWriter( neo4jConfig ) );
+        Map<Setting,Configuration> confNames = Configuration.getConfigurationNameMap();
+        writer.write( format( "%s=%s", confNames.get( Setting.HTTP_LISTEN_ADDRESS ).name, listenAddress ) );
+        writer.close();
+        temporaryFolderManager.mountHostFolderAsVolume( container, tempConfigDir, "/conf" );
+
+        return container;
+    }
+
+    @Test
+    void testContainerIsHealthyWhenNeo4jIsListeningAtPort7474()
+    {
+        try ( var container = createContainerWithDefaultListenAddress() )
+        {
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource( strings = {":4747", "127.0.0.1:4747", "localhost:4747", "localhost"} )
+    void testContainerIsHealthyWhenListenAddressIsModifiedByUser( String listenAddress )
+    {
+        try ( var container = createContainerWithEnvVarListenAddress( listenAddress ) )
+        {
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource( strings = {":4747", "127.0.0.1:4747", "localhost:4747", "localhost"} )
+    void testContainerIsHealthyWhenConfigIsModifiedByMounting( String listenAddress ) throws IOException
+    {
+        try ( var container = createContainerWithConfigListenAddress( listenAddress ) )
+        {
+            container.setWaitStrategy( Wait.forHealthcheck() );
+            container.start();
+
+            Assertions.assertTrue( container.isRunning() );
+            Assertions.assertEquals( "healthy", container.getCurrentContainerInfo().getState().getHealth().getStatus() );
+        }
+    }
```

Comment:
These are all great, we just need one more test.
Test when the listen address is set in neo4j.conf AND through environment variable.
In that situation we should expect the correct listen address to be whatever we set through env.